src/commands/use.rs

```diff
@@ -61,26 +60,9 @@ impl Command for Use {
                         .join("installation")
                 }
                 None => {
-                    ensure!(
-                        self.install_if_missing || should_install_interactively(&requested_version),
-                        CantFindVersion {
-                            version: requested_version
-                        }
-                    );
-
-                    Install {
-                        version: Some(requested_version.clone()),
-                        ..Default::default()
-                    }
-                    .apply(config)
-                    .context(InstallError)?;
-
-                    Self {
-                        version: Some(UserVersionReader::Direct(requested_version)),
-                        install_if_missing: self.install_if_missing,
-                    }
-                    .apply(config)?;
-
+                    let can_install =
+                        self.install_if_missing || should_install_interactively(&requested_version);
```

Comment:
can we move the `should_install_interactively` to be a part of the extracted function?
this is for suggesting the user `hey, this is missing, wanna install this?` -- and not just waiting for the `--install-if-missing` flag :)

Reply:
Other than that, it looks reaaaaaaally good. I hope it was easy figuring out how to write the e2e tests :)

Reply:
I made the change ;)  Yes, it took me a while to figure out how to create the snapshots. :D